### PR TITLE
clean rm unnecessary row info

### PR DESCRIPTION
**Pull request by alankagiri**

### CODE REVIEW

**Comment by trevorspreadbury** [comment]
looks like there's a few small pre-commit failures right now

File: notebooks/Test.ipynb
Comment (averyschoen) [inline notebooks/Test.ipynb:1]: delete this

File: utils/linkage.py
Comment (averyschoen) [inline utils/linkage.py:1]: I'm only going to comment on the duplicates->dict function -- see the other open PR for feedback on the name functions...

File: utils/linkage.py [hunk 1]
```diff
@@ -135,6 +270,77 @@ def get_street_from_address_line_1(address_line_1: str) -> str:
     return " ".join(string)
 
 
+def convert_duplicates_to_dict(df: pd.DataFrame) -> pd.DataFrame:
```
Comment (averyschoen) [inline utils/linkage.py]: this indicates that there is an output which conflicts with the docstring

File: utils/linkage.py [hunk 2]
```diff
@@ -135,6 +270,77 @@ def get_street_from_address_line_1(address_line_1: str) -> str:
...
+    """Saves to the "output" directory a file mapping multiple strings to one
+    string
+
+    Given a dataframe where each row contains one string in a column and a list
```
Comment (averyschoen) [inline utils/linkage.py:280]: While I think I understand what this function does, I don't think this explanation articulates it...

File: utils/linkage.py [hunk 3]
```diff
@@ -135,6 +270,77 @@ def get_street_from_address_line_1(address_line_1: str) -> str:
...
+    new_df = (
+        new_df.groupby(df.columns[1:].tolist(), dropna=False)["id"]
+        .agg(list)
+        .reset_index()
+        .rename(columns={"id": "duplicated"})
+    )
+    new_df.index = new_df["duplicated"].str[0].tolist()
+    new_df["duplicated"] = new_df["duplicated"].str[1:]
+
+    # now convert the duplicated column into a dictionary that can will be
```
Comment (averyschoen) [inline utils/linkage.py:337]: You could feed everything into this function even if the id was found to have no duplicates

File: utils/linkage.py [hunk 4]
```diff
@@ -135,6 +270,77 @@ def get_street_from_address_line_1(address_line_1: str) -> str:
...
+    Args:
+        a pandas dataframe containing contribution data
+    Returns:
+        a deduplicated pandas dataframe containing contribution data
+    """
+    # first remove all duplicate entries:
+    new_df = df.drop_duplicates()
+
+    # now find the duplicates along all columns but the ID
+    new_df = (
```
Comment (averyschoen) [inline utils/linkage.py:329]: bring this to slack -- how are null values handled?

File: utils/linkage.py [hunk 5]
```diff
@@ -135,6 +270,77 @@ def get_street_from_address_line_1(address_line_1: str) -> str:
...
+        deduped_uudis = df.iloc[i]["duplicated"]
+        for j in range(len(deduped_uudis)):
+            deduped_dict.update({deduped_uudis[j]: df.iloc[i]["id"]})
+
+    # now convert dictionary into a csv file
+    deduped_df = pd.DataFrame.from_dict(deduped_dict, "index")
+    deduped_df = deduped_df.reset_index().rename(
+        columns={"index": "duplicated_uuids", 0: "mapped_uuids"}
+    )
+    deduped_df.to_csv(
```
Comment (averyschoen) [inline utils/linkage.py:304]: if you add in the arg "header=false" I think this resolved the duplicated column name rows

**Review by averyschoen — CHANGES_REQUESTED** [review]
see in-line comments

File: utils/linkage.py (hunk 1 above)
Comment (alankagiri) [reply utils/linkage.py]: How did you even catch that that's amazing, thank you!

File: utils/linkage.py (hunk 3 above)
Comment (alankagiri) [reply utils/linkage.py:337]: I'm not sure what you mean by this comment @averyschoen

**Review by averyschoen — APPROVED** [review]

**Pull request merged**